### PR TITLE
backport: Merge bitcoin#21029, 21060, 19509, 21107, 25117

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

**Comment by vijaydasmp** [comment]
will resume after https://github.com/dashpay/dash/pull/5833 merges

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
LGTM but 21107 should be partial because we don't have 19509 yet

**Comment by UdjinM6** [comment]
pls see 7e47db12472a132905bd80aefa0ddff0deeaf546, 815aaecf5b411b47637451f31a942c37afea5f0b and b7fec6c0376969a4280591bcef7c339ffd4fedbc

**Comment by vijaydasmp** [comment]
> pls see [7e47db1](https://github.com/dashpay/dash/commit/7e47db12472a132905bd80aefa0ddff0deeaf546), [815aaec](https://github.com/dashpay/dash/commit/815aaecf5b411b47637451f31a942c37afea5f0b) and [b7fec6c](https://github.com/dashpay/dash/commit/b7fec6c0376969a4280591bcef7c339ffd4fedbc)

Thanks @UdjinM6 , I will apply this changes

File: src/net.cpp [hunk 1]
```diff
@@ -3986,6 +3986,9 @@ void CConnman::PushMessage(CNode* pnode, CSerializedNetMsg&& msg)
 {
     size_t nMessageSize = msg.data.size();
     LogPrint(BCLog::NET, "sending %s (%d bytes) peer=%d\n", SanitizeString(msg.command), nMessageSize, pnode->GetId());
+    if (gArgs.GetBoolArg("-capturemessages", false)) {
+        CaptureMessage(pnode->addr, msg.command, msg.data, /* incoming */ false);
+    }
```
Comment (UdjinM6) [inline src/net.cpp:3991]: 19509: linter complains
```suggestion
    }
```

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta , requesting review

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta , Please review

**Pull request merged**